react-ui/src/styleTransfer.js

```diff
@@ -0,0 +1,60 @@
+import * as tf from '@tensorflow/tfjs'
+tf.ENV.set('WEBGL_PACK', false)
+
+let styleModel, transferModel
+
+const styleModelPath = 'tf_models/style/model.json'
+const transferModelPath = 'tf_models/transfer/model.json'
+
+export default async function transformImageWithStyle(imageData, style) {
+
+  transferModel = transferModel || await tf.loadGraphModel(transferModelPath);
+
+  const stylized = await tf.tidy(() => {
+    const styleInput = styleFromArray(style)
+    const imgInput = tf.browser.fromPixels(imageData).toFloat().expandDims()
+    return transferModel.predict([imgInput, styleInput]).squeeze().div(tf.scalar(255.0))
+  })
+  return new ImageData(await tf.browser.toPixels(stylized),imageData.width,imageData.height) // returns Uint8ClampedArray
+}
+
+export async function getImageStyle(imageData) {
+
+  styleModel = styleModel || await tf.loadGraphModel(styleModelPath)
+
+  return await tf.tidy(() => {
+    const imgInput = tf.browser.fromPixels(imageData).toFloat().expandDims()
+    var res = styleToArray(styleModel.predict(imgInput).expandDims())
+    console.log(res)
+    return res
+  })
+}
+
+// Load image data from file (and shrink, if needed)
+export async function imageDataFromFile(file, max_width) {
```

Comment:
In large part, this duplicates and complicates what's in applyFilter indicating there's some common themes here.

I'd start from extract function here - function given an url loads the image data onto canvas and returns the data url.

If needed, the callee here would take care of scaling.

max_width in particular seems out of place for this function - we should handle scaling the image if needed else.